crates/wallet/src/types/node_url.rs

```diff
@@ -103,6 +107,18 @@ impl From<&NodeUrl> for tonic::transport::Endpoint {
     }
 }
 
+impl ToSql for NodeUrl {
+    fn to_sql(&self) -> Result<ToSqlOutput<'_>> {
+        Ok(self.as_ref().into())
+    }
+}
+
+impl FromSql for NodeUrl {
+    fn column_result(value: ValueRef<'_>) -> FromSqlResult<Self> {
+        String::column_result(value).map(NodeUrl::new_unchecked)
+    }
+}
+
```

Comment:
ideally behind a sqlite feature flag too. If we really want to be coherent across the whole codebase